docs/features/software-catalog/well-known-annotations.md

```diff
@@ -100,6 +100,21 @@ alongside the entity's source code, the value of this annotation can point to an
 absolute URL, matching the location reference string format outlined above, for
 example: `url:https://github.com/backstage/backstage/tree/master`
 
+### backstage.io/techdocs-ref
+
+```yaml
+# Example:
+metadata:
+  annotations:
+    backstage.io/techdocs-external-ref: component:default/example
```

Comment:
Is the annotation name at risk of being confusing? This "ref" is very unlike the other "ref" above. Other suggestions:

- `backstage.io/documented-by`: make this not about techdocs specifically, but as a general way of pointing elsewhere
- `backstage.io/techdocs-entity`
- keep using `backstage.io/techdocs-ref` with a different target type - like `backstage.io/techdocs-ref: entity:component:/default:foo`
- ...more?

What if were possible to just say `backstage.io/documented-by: system` as a shorthand 🤔 referring to your own system without having to name it fully. Or indeed `backstage.io/techdocs-ref: system:*` or whatever

Reply:
Sure I can do that I've gone with the long form to be explicit, as it might not be a system we know developers can be lazy and then you would have to track the relations causing a catalog call when building it. 

I figured it would require being opinionated how people structure their projects, and it might be confusing what if the system is several layers up so you might have System at the top, and this might be a subcomponent of a subcomponent.  This could also be a repo of subcomponents and not in a system. I figured if you are already writing system the specific system wouldn't be too much harder than the risk of picking the wrong system. 

The reason for leaving it outside the existing techdocs-ref, is it allows this change to be small and non confusing. inside the techdocs ref it would mean adding filtering to techdocs pages and a few other parts. This is explicitly in the case if you want the techdocs to be run by another component

I like the suggestion of `techdocs-entity` and have changed so that message as again its explicitly pointing at an entity